src/Symfony/Component/Form/Tests/Extension/Core/Type/ColorTypeTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Extension\Core\Type;
+
+use Symfony\Component\Form\Extension\Core\Type\ColorType;
+use Symfony\Component\Form\FormError;
+
+final class ColorTypeTest extends BaseTypeTest
+{
+    const TESTED_TYPE = ColorType::class;
+
+    /**
+     * @dataProvider validationShouldPassProvider
+     */
+    public function testValidationShouldPass(bool $html5, ?string $submittedValue)
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, [
+            'html5' => $html5,
+            'trim' => true,
+        ]);
+
+        $form->submit($submittedValue);
+
+        $this->assertEmpty($form->getErrors());
+    }
+
+    public function validationShouldPassProvider()
+    {
+        return [
+            [false, 'foo'],
+            [false, null],
+            [false, ''],
+            [true, '#000000'],
```

Comment:
Please add a test for `[true, '']`, `[true, ' ']` and `[true, null]` that should also pass unless using `NotBlank` proper validation.

Reply:
If I remember it correctly, the spec specify that empty values are impossible / invalid. Browsers always have to send a valid color, ie input type="color" always have a value. So if we allow empty string or null, we don't respect the spec.

Reply:
> User agents must not allow the user to set the value to the empty string.

>  If the value of the element is a valid simple color, then set it to the value of the element in ASCII lowercase; otherwise, set it to the string "#000000".

Maybe we can apply the same fallback to be defensive?

Reply:
From a component point of view we don't care about client side validation (i.e the html can be changed to `type="text"` before submit).
BUT either way, we must be able to use an invalid state to do validation, AND to use optional values.
Therefore, allowing to be strict AND optional, the following should be possible:
```php
$builder
    ->add('optional_color', ColorType::class, [
        'required' => false,
        'html5' => true,
    ])
;

$form->submit(['optional_color' => '']);

// true === $form->isValid()
```

Reply:
Alright, that makes sense.